benchmarks/pull.py

```diff
@@ -0,0 +1,43 @@
+import os
+import shutil
+
+from benchmarks.base import BaseRemoteBench
+
+
+class PullBench(BaseRemoteBench):
+    repeat = 1
+    timeout = 12000
+
+    def setup(self):
+        super().setup()
+
+        self.gen("data", "cats_dogs")
+        self.dvc("add", "data", "--quiet")
+        self.dvc("push", "data", "--quiet")
+
+        shutil.rmtree("data")
+        shutil.rmtree(os.path.join(".dvc", "cache"))
+
+    def time_cats_dogs(self):
+        self.dvc("pull", "-j", "2", proc=True)
+
+
+class PullOnlyMissingBench(BaseRemoteBench):
+
+    repeat = 1
+    timeout = 12000
+
+    def setup(self):
+        super().setup()
+
+        self.gen("data", "large")
+        self.dvc("add", "data", "--quiet")
+        self.dvc("push", "data", "--quiet")
+
+        shutil.rmtree("data")
+        shutil.rmtree(os.path.join(".dvc", "cache"))
+
+        self.gen("data", "medium")
```

Comment:
Ah, I see :hmm:. I think it is going to differ since it needs to calculate the missing files and for that, it needs to do `hashes_exist` calls (which is the part we are actually testing) though this is not an urgent benchmark so I could defer it